### ISSUE TITLE
Brave News header missing when interrupting page load by tapping `Home` button

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#21833]
## Description 

Brave News header missing when interrupting page load by tapping `Home` button

This is a followup for https://github.com/brave/brave-browser/issues/21267#issuecomment-1075744341

## Steps to reproduce 
1. install `1.38.66`
2. launch Brave
3. enable Brave News via `Settings` -> `Brave News` -> `Show Brave News`
4. swipe down
5. tap on `Show Brave News`
6. load `jalopnik.com`
7. interrupt the above load by tapping on the `Home` icon on the bottom toolbar
8. look for the `Brave News` header

## Actual result 

The `Brave News` header is missing

https://user-images.githubusercontent.com/387249/159762733-4890be08-53f5-4634-bb7b-e54db19a0688.mp4

## Expected result

Regardless of the user interaction, when on Brave News and scrolling, I'd expect the header to show up

## Issue reproduces how often 

100% with timing/patience

## Device details
   - Install type (ARM, x86): `arm64`
   - Device type (Phone, Tablet, Phablet): `Google Pixel XL`
   - Android version: `9.0`
   
## Brave version

`1.38.66, Chromium 100.0.4896.46`

cc @alexsafe @mattmcalister @srirambv

**Comment by kjozwiak** [comment]
Reproduce on my `Pixel 6` running `Android 12` using `1.38.63 Chromium: 100.0.4896.46`. Basically used the original STR that @srirambv provided via https://github.com/brave/brave-browser/issues/21267#issue-1148188539 and could reproduce the problem of the Brave News settings bar not being displayed. There's a gap where it should be visible.

**Comment by timchilds** [comment]
@kjozwiak I can also reproduce this on my device (Samsung Galaxy A51 / Android 11). 

cc @anthonypkeane @bsclifton

**Comment by Uni-verse** [comment]
Verified on `Nightly 1.39.39`, waiting for uplift.

**Comment by kjozwiak** [comment]
@tapanmodh technically this should still be in the `1.39.x` milestone as the original PR landed in `1.39.x` and this wasn't uplifted yet. Once https://github.com/brave/brave-core/pull/12931 is merged into `1.38.x`, the release team will move the issue into the correct milestone, in this case, `1.38.x`. You can view/read the process via https://github.com/brave/brave-browser/wiki/Triage-Guidelines#uplifting.

**Comment by kjozwiak** [comment]
https://github.com/brave/brave-core/pull/12817 was verified on Nightly as per https://github.com/brave/brave-core/pull/12817#issuecomment-1092426388 and uplifted via https://github.com/brave/brave-core/pull/12931.

Above requires `1.38.86` or higher for `1.38.x` verification.

**Comment by Uni-verse** [comment]
Verified using `1.38.99, Chromium 100.0.4896.127`

Devices: 
`Samsung Galaxy S21 | Android 12`
`Samsung Galaxy Tab S7 | Android 12`

- Was not able to reproduce this using above build version with following STR - https://github.com/brave/brave-browser/issues/21833#issue-1178471359
